QuantLib/ql/processes/blackscholesprocess.hpp

```diff
@@ -87,12 +91,14 @@ namespace QuantLib {
         Handle<BlackVolTermStructure> blackVolatility_;
         mutable RelinkableHandle<LocalVolTermStructure> localVolatility_;
         mutable bool updated_;
+        mutable boost::shared_ptr<LocalVolCurve> localVolCurve_;
+        mutable boost::shared_ptr<LocalConstantVol> localConstantVol_;
```

Comment:
yes, that's good. I will do it and update the PR.

On 15 May 2015 at 16:10, Luigi Ballabio <notifications@github.com> wrote:

> In QuantLib/ql/processes/blackscholesprocess.hpp
> <https://github.com/lballabio/quantlib/pull/234#discussion_r30410998>:
>
> > @@ -87,12 +91,14 @@ namespace QuantLib {
> >          Handle<BlackVolTermStructure> blackVolatility_;
> >          mutable RelinkableHandle<LocalVolTermStructure> localVolatility_;
> >          mutable bool updated_;
> > +        mutable boost::shared_ptr<LocalVolCurve> localVolCurve_;
> > +        mutable boost::shared_ptr<LocalConstantVol> localConstantVol_;
>
> Instead of storing these two (which forced you to add public variance
> methods to the LocalConstantVol and LocalVolCurve classes), what do you
> think about storing pointers (either shared_ or weak_) to the underlying
> black volatility instead, and calculate the variance in the evolve method?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lballabio/quantlib/pull/234/files#r30410998>.
>